src/theia/sfm/feature_extractor_and_matcher.cc

```diff
@@ -125,6 +161,27 @@ bool FeatureExtractorAndMatcher::AddImage(
   return true;
 }
 
+bool FeatureExtractorAndMatcher::SetMasksForFeaturesExtraction(
```

Comment:
This method is very hacky and seems potentially prone to error.

What about a different interface: bool AddMaskForFeatureExtraction(const std::string& image_filepath, const std::string& mask_filepath);

This method would then populate an internal std::unordered_map<std::string, std::string> image_masks_  which maps image_filepath --> mask_filepath. This way, there is never an ambiguity between what mask goes to which image.